source/core_mqtt_agent.c

```diff
@@ -980,6 +1002,15 @@ MQTTStatus_t MQTTAgent_Init( MQTTAgentContext_t * pMqttAgentContext,
                                   mqttEventCallback,
                                   pNetworkBuffer );
 
+        if( returnStatus == MQTTSuccess )
+        {
+            returnStatus = MQTT_InitStatefulQoS( &( pMqttAgentContext->mqttContext ),
```

Comment:
would if users didnt need QoS > 0? i suppose giving them that option may mean an API change... should be able to add this to the config tho, no?

Reply:
Yes, good point. Since we now have a config file, I added a config macro for this too.